sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -301,6 +301,38 @@ abstract class SchemaPruningSuite
     checkAnswer(query, Row("Y.", 1) :: Row("X.", 1) :: Row(null, 2) :: Row(null, 2) :: Nil)
   }
 
+  testSchemaPruning("select explode of nested field of array of struct") {
```

Comment:
I think the reason why we did not capture the bug is our tests are not well designed and reviewed. 

We have to be super careful when we review the tests and then it will be much easier to find the bugs.

Reply:
Thanks for catching it and pinging me. Let me look at it.

Reply:
Opened #27503 to fix it.